libvast/src/system/index.cpp

```diff
@@ -403,60 +403,60 @@ behavior index(stateful_actor<index_state>* self, const path& dir,
   // simply waits for a worker).
   self->set_default_handler(caf::skip);
   self->state.has_worker.assign(
-    [=](expression& expr) -> result<uuid, uint32_t, uint32_t> {
-      auto& st = self->state;
+    [=](expression& expr) {
+      auto respond = [&](auto&&... xs) {
+        auto mid = self->current_message_id();
+        unsafe_response(self, self->current_sender(), {}, mid.response_id(),
+                        std::forward<decltype(xs)>(xs)...);
```

Comment:
```suggestion
                        std::forward<std::decay_t<decltype(xs)>>(xs)...);
```

Reply:
Are you sure this is necessary? I've always seen it used without the decay.

Reply:
Assuming `auto&&` is the same type of forwarding reference as `T&&`, then you need to use the `std::decay`. Otherwise you'll get a reference via `decltype`. If you grep through the code base, you'll see many instances of this (unfortunately necessary) pattern.

Reply:
Apologies, I have to rudder back. Thanks for explaining this to me at the whiteboard in more detail, @dominiklohmann!

Reply:
We just concluded at the whiteboard that `decay` is incorrect here and should not be used.